File: solutions/ocaml/hamming/5/hamming.ml
type nucleotide = A | C | G | T ;;

let hamming_distance nuc1 nuc2 =
  match List.length nuc1, List.length nuc2 with
  | 0, _ -> Result.error "left strand must not be empty"
  | _, 0 -> Result.error "right strand must not be empty"
  | (x, y) when x == y -> List.fold_left (+) 0 ( List.map2 ( fun x y -> if x <> y then 1 else 0 ) nuc1 nuc2 ) |> Result.ok
  | _ -> Result.error "left and right strands must be of equal length"
;;

File: solutions/c/hamming/5/hamming.c
#include "hamming.h"

int compute(const char *left, const char *right) {

    if (strlen(left) != strlen(right)) {
        return -1;
    }
    
    if (strlen(left) == 0 || strlen(right) == 0) {
        return 0;
    }

    else {
        int distance = 0;
        int length = strlen(left);
        for (int i = 0; i < length; i++) {
            if (left[i] != right[i]) {
                distance++;
            }
        }
        return distance;
    }
}